src/DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapper.py

```diff
@@ -341,6 +351,11 @@ def execute(self):
       executable = executable.replace('$DIRACROOT', self.localSiteRoot)
       self.log.verbose('Replaced $DIRACROOT for executable as %s' % (self.localSiteRoot))
 
+    # Try to find the executable on PATH
+    if "/" not in executable:
+      # Returns None if the executable is not found so use "or" to leave it unchanged
+      executable = distutils.spawn.find_executable(executable) or executable
+
```

Comment:
Perhaps these 5 lines should backported to `rel-v7r1` so old pilots can still run new jobs?

Reply:
Yes, looks like a good idea.

Reply:
Done in https://github.com/DIRACGrid/DIRAC/pull/4998/